### PR TITLE
Ben/ner model logging efficiency spacy

### PR DESCRIPTION
**Pull request by Ben-Epstein**
original: https://github.com/rungalileo/dataquality/pull/519

 but with padding fixes for spacy, so we can utilize @rfriel  vectorization work

### CODE REVIEW

**Review by rfriel — APPROVED** [review]

File: dataquality/integrations/spacy.py [hunk 1]
```diff
@@ -524,11 +525,30 @@ def _self_get_valid_logits(self, docs: Dict[int, Doc]) -> DefaultDict:
...
+                pad_config = ((0, pad_size), (0, 0))
+                doc_embs = np.pad(doc_embs, pad_config, constant_values=0)
+                # For logits, we use our special logits to ensure it's an O prediction
+                doc_valid_logits = np.resize(
+                    doc_valid_logits, (max_doc_len, num_logits)
+                )
+                doc_valid_logits[-pad_size:, :] = o_logits
+
+            cast(List, model_logger.logits).append(np.array(doc_valid_logits))
+            cast(List, model_logger.embs).append(np.array(doc_embs))
```
Comment (elboy3) [inline dataquality/integrations/spacy.py:551]: is this just for linting? or does cast affect runtime

**Review by elboy3 — APPROVED** [review]

File: dataquality/integrations/spacy.py (hunk 1 above)
Comment (Ben-Epstein) [reply dataquality/integrations/spacy.py:551]: linting
